template.yaml

```diff
@@ -63,6 +67,40 @@ Resources:
                   - ses:SendRawEmail
                 Resource: "*"
 
+  DefaultLambdaRole:
+    Type: AWS::IAM::Role
+    Properties:
+      AssumeRolePolicyDocument:
+        Version: 2012-10-17
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service: [ lambda.amazonaws.com ]
+            Action: [ 'sts:AssumeRole' ]
+      Policies:
+        - PolicyName: writeLog
+          PolicyDocument:
+            Version: 2012-10-17
+            Statement:
+              - Effect: Allow
+                Action:
+                  - logs:CreateLogGroup
+                  - logs:CreateLogStream
+                  - logs:PutLogEvents
+                Resource: !Sub 'arn:aws:logs:${AWS::Region}:${AWS::AccountId}:log-group:*:*'
+        - PolicyName: sendMail
+          PolicyDocument:
+            Version: 2012-10-17
+            Statement:
+              - Effect: Allow
+                Action:
+                  - s3:GetObject
+                  - s3:ListBucket
+                  - s3:ListBucketMultipartUploads
+                  - s3:PutObject
+                  - s3:PutObjectAcl
+                Resource: "*"
```

Comment:
on all s3 buckets?

Reply:
Unfortunately the bucket we need access to is in another stack, so referencing it's ARN isn't simple. (I mean I could infer the ARN from the bucket name potentially, but I have no means of knowing if the bucket in the other stack is created before this rule is created).

Reply:
There is yet more tech debt we need to address: how to know if stack resources are created…

Reply:
You have the DependsOn keyword you can write in cloudformation yml.
Not sure if it is possible to use on resources from other stacks or even if we should do that.
We could end up creating circular dependency trees that won't be detected before we deploy our stack from a clean aws account.